### PR TITLE
Swift 5.6 support

### PR DESCRIPTION
**Pull request by erkekin**
addresses https://github.com/icanzilb/MarkCodable/issues/7

I removed non-collection branch of decode implementation, there was no tests covering that, let's discuss if that's still necessary. @icanzilb

### CODE REVIEW

**Comment by icanzilb** [comment]
Cool work @erkekin — if all current tests pass (admittedly, they're lacking in coverage but should be testing the basic use cases I could think of) I think we should be good with removing that code you cleaned.

My only concern is about the versioning, I have a separate issue for it https://github.com/icanzilb/MarkCodable/issues/15 

Since we can't seem to be able to have semantic versions and likely 5.6 vs 5.7 are going to be separate branches maybe we don't need the `#if #else #end` branching in code at all. I'll post a question re: semantic vs. branch versioning on the Swift forums but I think this shouldn't block us for now to merge this.

Do you have any thoughts regarding this? I'll have a moment later and I'll go over the changes in detail but looks great to me at first sight.

**Comment by erkekin** [comment]
I see, yeah I think that's a common practice to branch versioning when it comes to swift's beta versions, in this example 5.7. given it will be released soon we can refactor accordingly to swift to semantic versioning. I leave to you to give the decision, shall I remove if macros or merge this one now?

**Comment by icanzilb** [comment]
@erkekin I left a question here https://forums.swift.org/t/whats-the-recommended-way-to-depend-on-apple-packages-with-non-semantic-versioning/60125 and I think we shouldn't hold off merging code in. I'll figure out the dependencies later on.

File: Sources/MarkCodable/MarkCodable.swift
Comment (icanzilb) [inline Sources/MarkCodable/MarkCodable.swift:50]: Let's remove the `#if` and the fatal error here? The two code branches are anyways identical to:

```swift
extension Array where Element == Bool {
     mutating func push(_ value: Element) { append(value) }
     mutating func pop() { removeLast() }
 }
```

And that piece of code would work on both 5.6 and 5.7 without warnings. What do you think?

**Review by icanzilb — COMMENTED** [review]
Thanks so much for opening this! I left one question — let me know what do you think and after we'll be good to merge 👌🏼

File: Sources/MarkCodable/MarkCodable.swift
Comment (erkekin) [reply Sources/MarkCodable/MarkCodable.swift]: sure, I've just removed the redundant one and altered the `Bool` to `Appending` just in case. Thanks Marin.

**Review by icanzilb — APPROVED** [review]

**Comment by icanzilb** [comment]
Much appreciated, thank you

**Pull request merged**
